specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-03-01-preview/ExportConnections.json

```diff
@@ -0,0 +1,421 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "x-ms-examples": {
+          "Creates an export connection.": {
+            "$ref": "./examples/exportConnections/CreateExportConnection.json"
+          }
+        },
+        "tags": [
+          "ExportConnections"
+        ],
+        "description": "Creates an export connection.",
+        "operationId": "ExportConnections_Create",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/ExportConnectionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ExportConnection"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ExportConnection"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/ExportConnection"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ExportConnectionList": {
+      "description": "List all the export connections.",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of export connections.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of export connections.",
+          "items": {
+            "$ref": "#/definitions/ExportConnection"
+          },
+          "type": "array"
+        }
+      },
+      "required": [
+        "value"
+      ],
+      "type": "object"
+    },
+    "ExportConnection": {
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
```

Comment:
>ResourceWithEtag [](http://example.com/codeflow?start=63&length=16)

General comment: this definition (as checked in) actually does not include an etag, which seems weird. If the API actually includes an etag, you should use the AzureEntityResource definition from v4 common types.

Reply:
The 2.0 definition of ResourceWithEtag does include etag in its properties. 
"ResourceWithEtag": {
      "allOf": [
        {
          "$ref": "../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
        }
      ],
      "description": "An azure resource object with an Etag property",
      "properties": {
        "etag": {
          "description": "Etag of the azure resource",
          "type": "string"
        }
      },
      "type": "object"
    },
We have used this definition before with several other similar resources in different services. I believe this would work.

Reply:
You are right, not sure how I missed that.